File: r/5.R
#!/usr/bin/env Rscript

sideA <- function(input) {
    isNice <- function(str) {
        vowels <- '(.*[aeiou]){3}'
        substrings <- '(ab|cd|pq|xy)'
        doubleLetter <- '(.)\\1'

        grepl(vowels, str) & !grepl(substrings, str) & grepl(doubleLetter, str)
    }

    sum(isNice(input))
}

sideB <- function(input) {
    isNice <- function(str) {
        pair <- '(..).*\\1'
        repeater = '(.).\\1'

        grepl(pair, str, perl = TRUE) & grepl(repeater, str)
    }

    sum(isNice(input))
}

main <- function() {
    input <- readLines('../5.txt')
    cat(sideA(input), '\n')
    cat(sideB(input), '\n')
}

main()

File: js/5.js
#!/usr/bin/env node
'use strict'

const fs = require('fs')

const parse = (str) => str.split('\n')

const sideA = (arr) => arr.filter((str) => {
  const vowels = /(.*[aeiou]){3}/
  const substrings = /(ab|cd|pq|xy)/
  const doubleLetter = /(.)\1/

  return str.match(vowels) && !str.match(substrings) && str.match(doubleLetter)
}).length

// function sideB (arr) {
const sideB = (arr) => arr.filter((str) => {
  const pair = /(..).*\1/
  const repeater = /(.).\1/

  return str.match(pair) && str.match(repeater)
}).length

const main = () => {
  const contents = fs.readFileSync('../5.txt', 'utf8')
  const input = parse(contents)
  console.log(sideA(input))
  console.log(sideB(input))
  return
}

main()
